snow/validators/logger.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2019-2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package validators
+
+import (
+	"go.uber.org/zap"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/avalanchego/utils/set"
+)
+
+var _ SetCallbackListener = (*logger)(nil)
+
+type logger struct {
+	log      logging.Logger
+	enabled  *utils.Atomic[bool]
+	subnetID ids.ID
+	nodeIDs  set.Set[ids.NodeID]
+}
+
+// NewLogger returns a callback listener that will log validator set changes for
+// the specified validators
+func NewLogger(
+	log logging.Logger,
+	enabled *utils.Atomic[bool],
+	subnetID ids.ID,
+	nodeIDs ...ids.NodeID,
```

Comment:
Seems we only do this for a single NodeID. I guess this is just keeping the door open for more complex loggers?

Reply:
Yeah - just thought using the set was cleaner